crystal_diffusion/models/optimizer.py

```diff
@@ -0,0 +1,41 @@
+import logging
+from dataclasses import dataclass
+from enum import Enum
+
+import torch
+from torch import optim
+
+logger = logging.getLogger(__name__)
+
+
+class ValidOptimizerNames(Enum):
+    """Valid optimizer names."""
+    adam = "adam"
+    sgd = "sgd"
+
+
+@dataclass(kw_only=True)
+class OptimizerParameters:
+    """Parameters for the optimizer."""
+    name: ValidOptimizerNames
+    learning_rate: float
+
+
+def load_optimizer(hyper_params: OptimizerParameters, model: torch.nn.Module) -> optim.Optimizer:
+    """Instantiate the optimizer.
+
+    Args:
+        hyper_params : hyperparameters defining the optimizer
+        model : A neural network model.
+
+    Returns:
+        optimizer : The optimizer for the given model
+    """
+    match hyper_params.name:
+        case ValidOptimizerNames.adam:
+            optimizer = optim.Adam(model.parameters(), lr=hyper_params.learning_rate)
```

Comment:
depending on the regime, we might have to change the eps variable in adam. It might be premature to add it at this stage, it is easy enough to do later. Just something to keep in mind.